server/resolvers/book/query.js

```diff
@@ -0,0 +1,90 @@
+const { Op } = require("sequelize");
+const { Author, Book } = require("../../models");
+const { Metadata } = require("../../models/mongodb");
+
+const BookQuery = {
+  books: async ({ page = 1, limit = 10, filter = {} }) => {
+    const offset = (page - 1) * limit;
+
+    let whereClause = {};
+    let authorWhereClause = {};
+
+    if (filter.title) {
+      whereClause.title = { [Op.iLike]: `%${filter.title}%` };
+    }
+
+    if (filter.authorName) {
+      authorWhereClause.name = { [Op.iLike]: `%${filter.authorName}%` };
+    }
+
+    if (filter.publishedAfter) {
+      whereClause.published_date = { [Op.gte]: filter.publishedAfter };
+    }
+
+    if (filter.publishedBefore) {
+      whereClause.published_date = {
+        ...whereClause.published_date,
+        [Op.lte]: filter.publishedBefore,
+      };
+    }
```

Comment:
Anyway to make then reusable? Thinking this would be used from other queries as well?

Reply:
Yes your correct probably should have a utlity for pagination functionality.